mixer/adapter/solarwinds/config/config.proto

```diff
@@ -0,0 +1,129 @@
+option (gogoproto.gostring_all) = false;
+
+// Example config usage:
+// ```
+// apiVersion: "config.istio.io/v1alpha2"
+// kind: solarwinds
+// metadata:
+//   name: handler
+//   namespace: istio-system
+// spec:
+//   appoptics_access_token: <APPOPTICS SAMPLE TOKEN>
+//   papertrail_url: <PAPERTRAIL URL>
+//   papertrail_local_retention_duration: <RETENTION PERIOD FOR LOGS LOCALLY, Optional>
+//   metrics:
+//     requestcount.metric.istio-system:
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//       - response_code
+//     requestduration.metric.istio-system:
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//       - response_code
+//     requestsize.metric.istio-system:
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//       - response_code
+//     responsesize.metric.istio-system:
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//       - response_code
+//     tcpbytesent.metric.istio-system:
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//     tcpbytereceived.metric.istio-system:
+//       label_names:
+//       - source_service
+//       - source_version
+//       - destination_service
+//       - destination_version
+//   logs:
+//     solarwindslogentry.logentry.istio-system:
+//       payloadTemplate: '{{or (.originIp) "-"}} - {{or (.sourceUser) "-"}} [{{or (.timestamp.Format "2006-01-02T15:04:05Z07:00") "-"}}] "{{or (.method) "-"}} {{or (.url) "-"}} {{or (.protocol) "-"}}" {{or (.responseCode) "-"}} {{or (.responseSize) "-"}}'
+// ```
+message Params {
+    // AppOptics Access Token needed to send metrics to AppOptics. If no access token is given then metrics
+    // will NOT be shipped to AppOptics
+    string appoptics_access_token = 1;
+
+    // Optional. Max batch size of metrics to be sent to AppOptics.
+    // AppOptics does not allow batch size greater than 1000.
+    // If this is not specified a default batch size of 1000 will be used.
+    int32 appoptics_batch_size = 2;
+
+    // Papertrail url to ship logs to. If no papertrail url is given then the logs will NOT be shipped but rather
+    // dropped.
+    string papertrail_url = 3;
+
+    // This is the duration for which logs will be persisted locally until it is shipped to papertrail in the event
+    // of a network failure. Default value is 1 hour.
+    google.protobuf.Duration papertrail_local_retention_duration = 4 [(gogoproto.nullable)=true, (gogoproto.stdduration) = true];
+
+    // Describes how to represent an Istio metric in Solarwinds AppOptics
+    message MetricInfo {
```

Comment:
Hi @girishranganathan can you point me to the useage of the metricInfo object that is part of the adapter config ? I might be missing something here..

Reply:
It is just used for mapping metrics in the config and validation in the main solarwinds.go file.

Reply:
Since we are not using it (crtl+f on `.description` and `.labels` does not find anything), if we are not using MetricInfo object, should we just remove it completely ?